portal-bridge/src/beacon_bridge.rs

```diff
@@ -80,69 +79,83 @@ impl BeaconBridge {
     ///  Get and serve the latest beacon data.
     async fn launch_latest(&self) {
         // Current sync committee period known by the bridge
-        let current_period = Arc::new(AtomicUsize::new(0));
-        let finalized_block_root = Arc::new(Mutex::new(String::new()));
+        let (mut current_period, mut finalized_block_root) = Self::serve_latest(
+            self.api.clone(),
+            self.portal_clients.clone(),
+            0,
+            String::new(),
+        )
+        .await;
+
+        // Sleep until next update becomes available. This sets up the interval to update as soon as
+        // the following slot becomes available.
+        let now = SystemTime::now();
+        let next_update = duration_until_next_update(BEACON_GENESIS_TIME, now)
+            .to_std()
+            .expect("failed to convert chrono duration to std duration");
+        sleep(next_update).await;
+
+        // Run the beacon bridge update once every slot
+        let mut interval = interval(Duration::from_secs(12));
+        // If serving takes a little too long, then we want to serve the next one as soon as
+        // possible, but not serve any extras until the following slot. "Skip" gets this behavior.
+        interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
 
         loop {
             let consensus_api = self.api.clone();
             let portal_clients = self.portal_clients.clone();
-            let current_period = Arc::clone(&current_period);
-            let finalized_block_root = Arc::clone(&finalized_block_root);
-
-            tokio::spawn(async move {
-                Self::serve_latest(
-                    consensus_api,
-                    portal_clients,
-                    current_period,
-                    finalized_block_root,
-                )
-                .await;
-            });
 
-            let now = SystemTime::now();
+            (current_period, finalized_block_root) = Self::serve_latest(
+                consensus_api,
+                portal_clients,
+                current_period,
+                finalized_block_root,
+            )
+            .await;
 
-            let next_update = duration_until_next_update(BEACON_GENESIS_TIME, now)
-                .to_std()
-                .expect("failed to convert chrono duration to std duration");
-            sleep(next_update);
+            interval.tick().await;
         }
     }
 
     /// Serve latest beacon network data to the network
+    ///
+    /// Returns the new current period and finalized block root
     async fn serve_latest(
         api: ConsensusApi,
         portal_clients: Arc<Vec<HttpClient>>,
-        current_period: Arc<AtomicUsize>,
-        finalized_block_root: Arc<Mutex<String>>,
-    ) {
+        current_period: usize,
```

Comment:
nit: No need anymore to use `usize` for the period value, we can change it to `u64`.